### ISSUE TITLE
Kafka (redpanda) topics broken when consuming specific partition

### ISSUE THREAD

**Issue by parisholley** [Jeffail/benthos#715]
`Mar 27 22:32:42 redpanda_east_0 rpk[1025740]: INFO  2021-03-27 22:32:42,228 [shard 1] kafka - connection_context.cc:134 - Detected error processing request: std::runtime_error (Unsupported version 0 for offset commit API)
Mar 27 22:32:42 redpanda_east_0 rpk[1025740]: INFO  2021-03-27 22:32:42,841 [shard 1] kafka - connection_context.cc:134 - Detected error processing request: std::runtime_error (Unsupported version 0 for offset fetch API)`

using the legacy syntax works fine:

```yaml
topics: events
partition:1
```

but new syntax throws above errors

```yaml
topics: [ events:1 ]
```

**Comment by parisholley** [comment]
actually, the issue may be related to batching... i have two topics i am monitoring, so I tried breaking up using the legacy syntax like so:

```yaml
input:
  broker:
    batching:
      period: 1s
    inputs:
      - kafka:
          partition: 1
          topic: alpha
          start_from_oldest: false

      - kafka:
          partition: 1
          topic: beta
          start_from_oldest: false
```

the above caused the same problem, but moving batching to input level fixed it:

```yaml
input:
  broker:
    inputs:
      - kafka:
          partition: 1
          topic: alpha
          start_from_oldest: false
          batching:
            period: 1s

      - kafka:
          partition: 1
          topic: beta
          start_from_oldest: false
          batching:
            period: 1s
```

**Comment by Jeffail** [comment]
Hey @parisholley, I can reproduce this but looks like I'll need to do a bit of digging around, the error seems to be with committing offsets when consuming explicit partitions. It seems like the error occurs with both the legacy and new versions, but the difference is that the new version attempts more frequently so you'll see errors more often.

**Comment by Jeffail** [comment]
Got it: https://github.com/Jeffail/benthos/commit/36c34c3c92b49eaa13733e333f51f76bd74b8226
Benthos was using the lowest API version available (v0.8.1) for offset commits, which Redpanda doesn't support. I've increased it to v0.8.2 and it seems to work now with the latest Redpanda image.

**Comment by parisholley** [comment]
@Jeffail awesome, thanks for fixing so fast! **sponsors**
